src/AutoRest.CSharp/MgmtTest/Generation/ScenarioTestWriter.cs

```diff
@@ -0,0 +1,347 @@
+
+                var existedVariables = testDef.GetAllVariableNames();
+                using (new CodeWriterOavVariableScope(writer, scenario, existedVariables))
+                {
+                    WriteVariableInitializations(scenario, existedVariables);
+                    writer.Line($"resourceGroupName = Recording.GenerateAssetName(resourceGroupName);");
+                    writer.Line($"resourceGroup = (await GetArmClient().GetSubscription(new ResourceIdentifier($\"/subscriptions/{{subscriptionId}}\")).GetResourceGroups().CreateOrUpdateAsync(true, resourceGroupName, new ResourceGroupData(new AzureLocation(location)))).Value;");
+
+                    foreach (var step in scenario.Steps)
+                    {
+                        using (writer.Scope())
+                        {
+                            WriteStep(scenario, step);
+                        }
+                        writer.Line();
+                    }
+                }
+            }
+            writer.Line();
+        }
+
+        protected void WriteVariableInitializations(OavVariableScope ovs, IEnumerable<string> existedVarialbes)
+        {
+            IEnumerable<string> requiredVariables = ovs.RequiredVariables ?? new List<string>();
+            Dictionary<string, string> requiredVariableDefault = ovs.RequiredVariablesDefault ?? new Dictionary<string, string>();
+            var variables = ovs.Variables ?? new Dictionary<string, object?>();
+
+            if (ovs is TestDefinitionModel)
+            foreach (var requiredVariable in requiredVariables)
+            {
+                if (!existedVarialbes.Contains(requiredVariable))
+                {
+                    writer.Append($"string {new CodeWriterDeclaration(requiredVariable):D} = Environment.GetEnvironmentVariable(\"{requiredVariable.ToSnakeCase().ToUpper()}\")");
+                }
+                else
+                {
+                    writer.Append($"{requiredVariable} = Environment.GetEnvironmentVariable(\"{requiredVariable.ToSnakeCase().ToUpper()}\")");
+                }
+                if (requiredVariableDefault.ContainsKey(requiredVariable))
+                {
+                    writer.Append($" ?? {requiredVariableDefault[requiredVariable]:L}");
+                }
+                writer.Line($";");
+            }
+
+            foreach (var variable in variables.Keys)
+            {
+                var defaultValue = ovs.GetVariableDefaultValue(variable).RefScenariDefinedVariables(existedVarialbes);
+                if (!requiredVariables.Contains(variable) && !existedVarialbes.Contains(variable))
+                {
+                    writer.Line($"string {new CodeWriterDeclaration(variable):D} = {defaultValue};");
+                }
+                else
+                {
+                    writer.Line($"{variable} = {defaultValue};");
+                }
+            }
+
+            if (ovs is not TestDefinitionModel && variables.Count()>0)
```

Comment:
Unless the type of `variables` is `IEnumerable<>`, it should have a property `Count`. And this file needs a formating
```suggestion
            if (ovs is not TestDefinitionModel && variables.Count>0)
```